File: app/src/main/java/com/tohami/egyptinnovate/ui/navigation/NavigationDrawerFragment.java
package com.tohami.egyptinnovate.ui.navigation;


import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;

import com.tohami.egyptinnovate.R;


/**
 * A simple {@link Fragment} subclass.
 */
public class NavigationDrawerFragment extends Fragment {


    private final View.OnClickListener onClickListener;


    public NavigationDrawerFragment(View.OnClickListener onClickListener) {
        this.onClickListener = onClickListener ;
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_navigation_drawer, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        view.findViewById(R.id.drawer_events).setOnClickListener(onClickListener);
        view.findViewById(R.id.drawer_leadership).setOnClickListener(onClickListener);
        view.findViewById(R.id.drawer_language).setOnClickListener(onClickListener);
        view.findViewById(R.id.drawer_map).setOnClickListener(onClickListener);
        view.findViewById(R.id.drawer_news).setOnClickListener(onClickListener);
    }
}

File: app/src/main/java/com/tohami/egyptinnovate/ui/navigation/NavigationDrawerFragment.kt
package com.tohami.egyptinnovate.ui.navigation


import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.tohami.egyptinnovate.R
import kotlinx.android.synthetic.main.fragment_navigation_drawer.*


class NavigationDrawerFragment : Fragment() {


    private var onClickListener: View.OnClickListener? = null


    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_navigation_drawer, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        drawer_events.setOnClickListener(onClickListener)
        drawer_leadership.setOnClickListener(onClickListener)
        drawer_language.setOnClickListener(onClickListener)
        drawer_map.setOnClickListener(onClickListener)
        drawer_news.setOnClickListener(onClickListener)
    }
}
